manifests/init.pp

```diff
@@ -82,37 +82,39 @@
-  Optional[Variant[Array,String]]                        $ntp_server,
-  Optional[Variant[Array,String]]                        $fallback_ntp_server,
-  Boolean                                                $manage_accounting,
-  Hash[String,String]                                    $accounting,
-  Boolean                                                $purge_dropin_dirs,
-  Boolean                                                $manage_journald,
-  Systemd::JournaldSettings                              $journald_settings,
-  Boolean                                                $manage_logind,
-  Systemd::LogindSettings                                $logind_settings,
+  Optional[Hash[String,Hash[String, Any]]]               $service_limits      = undef,
+  Boolean                                                $manage_resolved     = false,
+  Enum['stopped','running']                              $resolved_ensure     = 'running',
+  Optional[Variant[Array[String],String]]                $dns                 = undef,
+  Optional[Variant[Array[String],String]]                $fallback_dns        = undef,
+  Optional[Variant[Array[String],String]]                $domains             = undef,
+  Optional[Variant[Boolean,Enum['resolve']]]             $llmnr               = undef,
+  Optional[Variant[Boolean,Enum['resolve']]]             $multicast_dns       = undef,
+  Optional[Variant[Boolean,Enum['allow-downgrade']]]     $dnssec              = undef,
+  Optional[Variant[Boolean,Enum['opportunistic', 'no']]] $dnsovertls          = undef,
+  Boolean                                                $cache               = false,
+  Optional[Variant[Boolean,Enum['udp','tcp']]]           $dns_stub_listener   = undef,
+  Boolean                                                $use_stub_resolver   = false,
+  Boolean                                                $manage_networkd     = false,
+  Enum['stopped','running']                              $networkd_ensure     = 'running',
+  Boolean                                                $manage_timesyncd    = false,
+  Enum['stopped','running']                              $timesyncd_ensure    = 'running',
+  Optional[Variant[Array,String]]                        $ntp_server          = undef,
+  Optional[Variant[Array,String]]                        $fallback_ntp_server = undef,
+  Boolean                                                $manage_accounting   = false,
+  Optional[Hash[String,String]]                          $accounting          = undef,
+  Boolean                                                $purge_dropin_dirs   = false,
+  Boolean                                                $manage_journald     = false,
+  Optional[Systemd::JournaldSettings]                    $journald_settings   = undef,
+  Boolean                                                $manage_logind       = false,
+  Optional[Systemd::LogindSettings]                      $logind_settings     = undef,
```

Comment:
Or maybe put the defaults in the module's hiera data.

Reply:
https://github.com/camptocamp/puppet-systemd/pull/136#issuecomment-572047779

Reply:
Yes, that's why I suggest to put it in the module's hiera data, so you don't have to put it in your site's hiera data.